### PR TITLE
[WFLY-19590] [WFLY-19797] [MP Platform 7] MicroProfile Telemetry 2.0

### PR DESCRIPTION
**Pull request by jasondlee**
https://issues.redhat.com/browse/WFLY-19590

Also https://issues.redhat.com/browse/WFLY-19797 via the change to testsuite/integration/microprofile-tck/pom.xml

- Update otel artifact names and versions for Preview
- Update BOMs to reflect split between Standard and Preview
- Duplicate models under Preview to update for 2.0
- Add (temporary) check for Preview to set up exports as appropriate
- Modify preview test profiles to test 2.0 instead of 1.1

### CODE REVIEW

File: galleon-pack/galleon-shared/src/main/resources/modules/system/layers/base/io/grpc/main/module.xml
Comment (bstansberry) [inline galleon-pack/galleon-shared/src/main/resources/modules/system/layers/base/io/grpc/main/module.xml:1]: This file remains unchanged and instead a new module needs to be created in preview/galleon-local.

File: galleon-pack/galleon-shared/src/main/resources/modules/system/layers/base/io/opentelemetry/instrumentation/api/main/module.xml
Comment (bstansberry) [inline galleon-pack/galleon-shared/src/main/resources/modules/system/layers/base/io/opentelemetry/instrumentation/api/main/module.xml:1]: This file remains unchanged and instead a new module needs to be created in preview/galleon-local.

**Review by bstansberry — CHANGES_REQUESTED** [review]
@jasondlee I made a bunch of comments all related to what's needed for this to be a WildFly Preview only feature, which per Friday discussion is the short term goal.

I didn't comment on the testsuite stuff because it wasn't clear to me if those changes required the production code changes, or if they'd work fine in the existing standard WF.

I also didn't comment on the couple System.out.println in the ts, because those are OT to my goal of discussing the 'how to do it in WFP only' topic.

File: boms/common-expansion/pom.xml
[the remark this answers is not in the document]
Comment (jasondlee) [reply boms/common-expansion/pom.xml:1]: I think have made all of the requested changes.

File: observability/opentelemetry/src/main/java/org/wildfly/extension/opentelemetry/OpenTelemetrySubsystemRegistrar.java [hunk 1]
```diff
@@ -59,6 +59,7 @@ class OpenTelemetrySubsystemRegistrar implements SubsystemResourceDefinitionRegi
     public static final String[] EXPORTED_MODULES = {
             "io.opentelemetry.api",
             "io.opentelemetry.api.events",
+            "io.opentelemetry.context",
```
[the remark this answers is not in the document]
Comment (jasondlee) [reply observability/opentelemetry/src/main/java/org/wildfly/extension/opentelemetry/OpenTelemetrySubsystemRegistrar.java:76]: @bstansberry  This change is not strictly necessary for WF Standard, but it also does not hurt. Unless you feel strongly, I'm going to leave this here for simplicity's sake, which means, I guess, the Class.forName() above is not necessary here.

[the remark this answers is not in the document]
Comment (bstansberry) [reply observability/opentelemetry/src/main/java/org/wildfly/extension/opentelemetry/OpenTelemetrySubsystemRegistrar.java:76]: OK. This is one that was exported in WF 33 anyway.

File: preview/galleon-local/src/main/resources/modules/system/layers/base/io/opentelemetry/api/main/module.xml [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    <dependencies>
+        <module name="io.opentelemetry.api.events"/>
+        <module name="io.opentelemetry.context"/>
+        <module name="io.opentelemetry.instrumentation.annotations"/>
+        <module name="io.opentelemetry.instrumentation.api"/>
+        <module name="io.opentelemetry.semconv"/>
+        <module name="io.smallrye.opentelemetry" services="export"/>
+
+        <module name="java.logging"/>
+        <module name="jdk.management"/>
```
Comment (bstansberry) [inline preview/galleon-local/src/main/resources/modules/system/layers/base/io/opentelemetry/api/main/module.xml:29]: See my comment at https://github.com/wildfly/wildfly/pull/18266#discussion_r1784844657

**Review by bstansberry — APPROVED** [review]

**Comment by bstansberry** [comment]
Thansk @jasondlee and @jamezp

**Pull request merged**